content/glossary/build_tool.md

```diff
@@ -53,4 +61,18 @@ flowchart LR
     compiler --> .pyc
 ```
 
-The totality of ancestors for a given artifact may be represented as an [artifact tree](/glossary/artifact_tree).
\ No newline at end of file
+The totality of ancestors for a given artifact may be represented as an [artifact tree](/glossary/artifact_tree).
+
+## Code Generators
+
+Typically, source code files are hand written by humans, and as such are [leaf artifacts](http://localhost:1313/glossary/artifact/#leaf-artifacts) in the [artifact tree](/glossary/artifact_tree).
+
+Sometimes source code files are themselves **generated** by a code generator from inputs.
+
+```mermaid
+flowchart LR
+    codegeninput[codegen input] --> codegenerator[[code generator]] --> generatedsrc[generated source code file]
+```
+
+A code generator is a [build tool](#build-tools).  The output from a [build tool](#build-tools) is a [derived artifact](http://localhost:1313/glossary/artifact/#derived-artifacts).  Therefore generated soure code is a [derived artifact](http://localhost:1313/glossary/artifact/#derived-artifacts).
```

Comment:
Suggest rewriting this to get to the point first, then explain why. E.g. 
```suggestion
In this scenario, generated source code is a [derived artifact](http://localhost:1313/glossary/artifact/#derived-artifacts). This is because the [code generator](build_tool/#code-generators) is a [build tool](#build-tools) and, by definition, the output from the [build tool](#build-tools) is a [derived artifact](http://localhost:1313/glossary/artifact/#derived-artifacts).
```

Reply:
Done.